Seravo/seravo-plugin
# Add config files checker to PHP version switcher

#### What are the main changes in this PR?
PHP Version Switcher:
1. Check current PHP file.
2. Disable PHP Switcher if the current config file is not custom.conf, examples.conf or php.conf.
3. If php.conf exists, comment out "set $mode php" in custom.conf and exampes.conf if these exists.

##### Why are we doing this? Any context or related work?
Number 2 because if the developer sets "set $mode php7.3" to xyz-version.conf, the PHP Version Switcher cannot change the PHP version afterward. 

Number 3 because wp-restart-nginx could look like:
```testing nginx configuration...
nginx:` the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf test is successful
Checking for PHP backends...
Found PHP mode definition: php7.2 in '/data/wordpress/nginx/custom.conf'
Found PHP mode definition: php7.3 in '/data/wordpress/nginx/examples.conf'
Found PHP mode definition: php7.3 in '/data/wordpress/nginx/php.conf'
WARNING: Found multiple PHP mode definitions (see above), using last one (php7.3)
PHP7.3 mode enabled
Restarting nginx...
Nginx restarted!
```

#### Manual testing steps?
1 & 2.
	Set "set $mode php7.3;" to /data/wordpress/nginx/xyz_test.php
	Reload updates page

3.
	Check that /data/wordpress/nginx does not contain any other files than: custom.conf, examples.conf and php.conf
	Check that php.conf and custom.conf or examples.conf contain uncommented "set $mode phpX.X;"
	Reload updates page
	Then check custom.conf or examples.conf. Now this file should contain "#set $mode phpX.X;"

#### Screenshots
![change_php_version_2](https://user-images.githubusercontent.com/43742796/61708901-a6476400-ad56-11e9-9266-43470dfbbe68.png)

![change-php-version](https://user-images.githubusercontent.com/43742796/61708909-ac3d4500-ad56-11e9-8c3f-1a41a79b6f93.png)
